embrace-gradle-plugin/src/main/java/io/embrace/android/gradle/plugin/tasks/ndk/CollectSharedObjectFilesTaskRegistration.kt

```diff
@@ -0,0 +1,112 @@
+package io.embrace.android.gradle.plugin.tasks.ndk
+
+import com.android.build.api.variant.Variant
+import io.embrace.android.gradle.plugin.config.PluginBehavior
+import io.embrace.android.gradle.plugin.config.ProjectType
+import io.embrace.android.gradle.plugin.config.UnitySymbolsDir
+import io.embrace.android.gradle.plugin.gradle.nullSafeMap
+import io.embrace.android.gradle.plugin.gradle.registerTask
+import io.embrace.android.gradle.plugin.gradle.safeFlatMap
+import io.embrace.android.gradle.plugin.gradle.tryGetTaskProvider
+import io.embrace.android.gradle.plugin.instrumentation.config.model.EmbraceVariantConfig
+import io.embrace.android.gradle.plugin.model.AndroidCompactedVariantData
+import io.embrace.android.gradle.plugin.tasks.registration.EmbraceTaskRegistration
+import io.embrace.android.gradle.plugin.tasks.registration.RegistrationParams
+import io.embrace.android.gradle.plugin.util.capitalizedString
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.TaskProvider
+import java.io.File
+
+class CollectSharedObjectFilesTaskRegistration(
+    private val behavior: PluginBehavior,
+    private val unitySymbolsDir: Provider<UnitySymbolsDir>,
+    private val projectType: Provider<ProjectType>,
+) : EmbraceTaskRegistration {
+    override fun register(params: RegistrationParams) {
+        params.execute()
+    }
+
+    fun RegistrationParams.execute(): TaskProvider<CollectSharedObjectFilesTask>? {
+        val variantConfig = variantConfigurationsListProperty.get().first { it.variantName == variant.name }
+        val embraceConfig = variantConfig.embraceConfig
+
+        if (embraceConfig?.ndkEnabled == false) return null
+
+        val collectSharedObjectFilesTaskProvider = project.registerTask(
+            CollectSharedObjectFilesTask.NAME,
+            CollectSharedObjectFilesTask::class.java,
+            data
+        ) { task ->
+            // TODO: is the ndkEnabled check needed? We are already checking that in the execute method above
+            task.onlyIf { shouldExecuteTask(embraceConfig) }
```

Comment:
Can you test the behavior of this? I think this depends on whether `shouldExecuteTask` relies on not being called early in the Gradle build lifecycle, or whether the information is all there as soon as the build initializes.

Reply:
just verified that the task won't be registrated if ndkEnabled isn't true, so this check is not necessary